scanners/zap-extended/scanner/scbzapv2/zap_spider.py

```diff
@@ -0,0 +1,373 @@
+            spiderId = self.__zap.ajaxSpider.scan(url=target, contextname=context_name)
+
+        return spiderId
+
+    def __configure_spider(self, zap_spider: spider, spider_config: collections.OrderedDict):
+        """ Starts a ZAP Spider with the given name for the spiders configuration, based on the given configuration and ZAP instance.
+        
+        Parameters
+        ----------
+        zap_spider: spider
+            The reference to the running ZAP spider to configure.
+        spider_config: collections.OrderedDict
+            The spider configuration based on ZapConfiguration.
+        """
+
+        logging.debug('Trying to configure the Spider')
+            
+        # Configure Spider (ajax or http)
+        
+        if "maxDuration" in spider_config and (spider_config['maxDuration'] is not None) and spider_config['maxDuration'] >= 0:
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_max_duration(str(spider_config['maxDuration'])), 
+                method="set_option_max_duration"
+            )
+        if "maxDepth" in spider_config and (spider_config['maxDepth'] is not None) and spider_config['maxDepth'] >= 0:
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_max_depth(str(spider_config['maxDepth'])), 
+                method="set_option_max_depth"
+            )
+        if "maxChildren" in spider_config and (spider_config['maxChildren'] is not None) and spider_config['maxChildren'] >= 0:
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_max_children(str(spider_config['maxChildren'])), 
+                method="set_option_max_children"
+            )
+        if "maxParseSizeBytes" in spider_config and (spider_config['maxParseSizeBytes'] is not None) and spider_config['maxParseSizeBytes'] >= 0:
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_max_parse_size_bytes(str(spider_config['maxParseSizeBytes'])), 
+                method="set_option_max_parse_size_bytes"
+            )
+        if "acceptCookies" in spider_config and (spider_config['acceptCookies'] is not None) :
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_accept_cookies(str(spider_config['acceptCookies'])), 
+                method="set_option_accept_cookies"
+            )
+        if "handleODataParametersVisited" in spider_config and (spider_config['handleODataParametersVisited'] is not None) :
+            self.__check_zap_spider_result(
+                spiderId=zap_spider.set_option_handle_o_data_parameters_visited(str(spider_config['handleODataParametersVisited'])), 
+                method="set_option_handle_o_data_parameters_visited"
+            )
+        if "handleParameters" in spider_config and (spider_config['handleParameters'] is not None) :
```

Comment:
This and the following might be re-written into a loop with key, method parmeter and method to call since they all seems to have the same signature.

Reply:
Good Point 🐍 🤖